include/cfdcore/cfdcore_ecdsa_adaptor.h

```diff
@@ -0,0 +1,175 @@
+   *
+   * @return ByteData
+   */
+  ByteData GetData() const;
+
+ private:
+  /**
+   * @brief The underlying data
+   *
+   */
+  ByteData data_;
+};
+
+/**
+ * @brief A proof for an adaptor signature.
+ *
+ */
+class CFD_CORE_EXPORT AdaptorProof {
+ public:
+  /**
+   * @brief The size of an adaptor proof.
+   *
+   */
+  static constexpr uint32_t kAdaptorProofSize = 97;
+  /**
+   * @brief Construct a new Adaptor Proof from ByteData
+   *
+   * @param data the data representing the proof
+   */
+  explicit AdaptorProof(const ByteData &data);
+  /**
+   * @brief Construct a new Adaptor Proof object from a string
+   *
+   * @param data the data representing the proof
+   */
+  explicit AdaptorProof(const std::string &data);
+
+  /**
+   * @brief Get the underlying ByteData object.
+   *
+   * @return ByteData
+   */
+  ByteData GetData() const;
+
+ private:
+  /**
+   * @brief The underlying data
+   *
+   */
+  ByteData data_;
+};
+
+/**
+ * @brief A pair of an adaptor signature together with the proof attached to it.
+ *
+ */
+struct CFD_CORE_EXPORT AdaptorPair {
+  /**
+   * @brief An adaptor signature
+   *
+   */
+  AdaptorSignature signature;
+  /**
+   * @brief An adaptor proof
+   *
+   */
+  AdaptorProof proof;
+};
+
+/**
+ * @brief This class contain utility functions to work with adaptor signatures.
+ */
+class CFD_CORE_EXPORT AdaptorUtil {
+ public:
+  /**
+   * @brief Create an adaptor signature over the given message using the given
+   * private key. Returns an AdaptorPair of the adaptor signature and its proof.
+   *
+   * @param msg the message to create the signature for.
+   * @param sk the secret key to create the signature with.
+   * @param adaptor the adaptor to adapt the signature with.
+   * @return AdaptorPair
+   */
+  static AdaptorPair Sign(
+      const ByteData256 &msg, const Privkey &sk, const Pubkey &adaptor);
+
+  /**
+   * @brief "Decrypt" an adaptor signature using the provided secret, returning
+   * an ecdsa signature in compact format.
+   *
+   * @param signature the adaptor signature
+   * @param adaptor_secret the secret
+   * @return ByteData
+   */
+  static ByteData Adapt(
+      const AdaptorSignature &signature, const Privkey &adaptor_secret);
+
+  /**
+   * @brief Extract an adaptor secret from an ECDSA signature for a given
+   * adaptor signature.
+   *
+   * @param adaptor_signature the adaptor signature
+   * @param signature the ECDSA signature
+   * @param adaptor the adaptor for the signature
+   * @return Privkey
+   */
+  static Privkey ExtractSecret(
+      const AdaptorSignature &adaptor_signature, const ByteData &signature,
+      const Pubkey &adaptor);
+
+  /**
+   * @brief Verify that an adaptor proof is valid with respect to a given
+   * adaptor signature, adaptor, message and public key.
+   *
+   * @param adaptor_signature
+   * @param proof
+   * @param adaptor
+   * @param msg
+   * @param pubkey
+   * @return true
+   * @return false
```

Comment:
In case of bool return, I think this is better.
```suggestion
   * @retval true
   * @retval false
```

Reply:
Changed